### PR TITLE
Upgrade example to latest react native 61.1

### PR DESCRIPTION
**Pull request by RobertSasak**
I knew that this PR will touch many files and it will be hard to review. So I created a _upgradeExample.sh_ script so that is possible to recreate my changes. Here is a copy of it
```sh
#!/bin/bash

react-native init RNMapboxGLExample
mv example/src RNMapboxGLExample/src
mv example/scripts RNMapboxGLExample/scripts
rm -rf example
mv RNMapboxGLExample example

# Edit package.json

cd example
npx json -I -f package.json -e 'this.scripts["copy:changes"]="node ./scripts/watch_rngl.js"'
npx json -I -f package.json -e 'this.scripts["pack:gl"]="./scripts/npm_pack_rngl.sh"'
npx json -I -f package.json -e 'this.scripts["clean:node:modules"]="./scripts/clean_node_modules.sh"'
npx json -I -f package.json -e 'this.scripts.preinstall="npm run pack:gl"'
npx json -I -f package.json -e 'this.scripts.postinstall="node ./scripts/set_access_token.js"'
npx json -I -f package.json -e 'this.scripts["reset:from:gl"]="npm run clean:node:modules && npm install"'
npx json -I -f package.json -e 'this.dependencies["@react-native-mapbox-gl/maps"]="file:../react-native-mapbox-gl-maps.tgz"'

# Install depencies
touch accesstoken
yarn add @mapbox/geo-viewport@0.4.0 @turf/along@5.1.5 @turf/bearing@5.1.5 @turf/distance@5.1.5 @turf/helpers@4.7.3 @turf/line-distance@4.7.3 @turf/nearest@4.7.3 buffer@5.1.0 install@0.12.2 @mapbox/mapbox-sdk@0.6.0 moment@2.24.0 npm@5.10.0 prop-types@15.7.2 react-native-elements@1.1.0 react-native-vector-icons react-native-safe-area-view@0.13.1 react-navigation@2.18.3 url@0.11.0
react-native link
rm accesstoken
cd ios && pod install && cd ../
```
In addition to this script I edited _index.js_
```js
+import App from './src/App';
-import App from './App';
```

### CODE REVIEW

**Comment by tsemerad** [comment]
It looks like CI is failing on the Android build:

```bash
* What went wrong:
Could not determine the dependencies of task ':app:lintVitalRelease'.
> Could not resolve all task dependencies for configuration ':app:releaseRuntimeClasspath'.
   > Could not resolve project :@react-native-mapbox-gl_maps.
     Required by:
         project :app
      > Unable to find a matching configuration of project :@react-native-mapbox-gl_maps:
          - None of the consumable configurations have attributes.
```

There's a workaround [here](https://github.com/facebook/react-native/issues/5752#issuecomment-203043427) disabling linting the release build using `lintOptions { checkReleaseBuilds false }`. I'm not sure if it's an ideal long-term solution, but I'm not sure if it makes sense to be linting the release build either.

**Comment by tsemerad** [comment]
I should have began with this - thanks for the PR! This is greatly appreciated. I don't know if @kristfal or @mfazekas have any suggestions other than the `lintOptions { checkReleaseBuilds false }` I mentioned, but I think that's worth a try.

**Comment by RobertSasak** [comment]
# First issue
How can I reproduce linter error at my macOS? I tried `/android$ ./gradlew assemble` but it compiled without warning. Here is an output from travis. Sounds like linter cannot find *@react-native-mapbox-gl_maps* project. Sounds important to me. I need to fix it.
```
* What went wrong:
Could not determine the dependencies of task ':app:lintVitalRelease'.
> Could not resolve all task dependencies for configuration ':app:releaseRuntimeClasspath'.
   > Could not resolve project :@react-native-mapbox-gl_maps.
     Required by:
         project :app
      > Unable to find a matching configuration of project :@react-native-mapbox-gl_maps:
          - None of the consumable configurations have attributes.
```

# Second issue
While it works on Android emulator, it crash on actual real device. Example app crash a second after app starts. I do not see any actual error in logcat. How one debug this kind of errors?
```
09-29 00:01:59.714 ? D/PowerManagerNotifier: onWakeLockReleased: flags=1, tag="com.mapbox.android.telemetry.crash.CrashReporterJobIntentService:run", packageName=com.rnmapboxglexample, ownerUid=10200, ownerPid=30674, workSource=null
09-29 00:01:59.767 ? I/ReactNativeJS: Running "RNMapboxGLExample" with {"rootTag":1}
09-29 00:02:00.323 ? I/BufferQueueProducer: [com.rnmapboxglexample/com.rnmapboxglexample.MainActivity](this:0x7f83bca800,id:1314,api:1,p:30674,c:265) new GraphicBuffer needed
09-29 00:02:00.329 ? I/BufferQueueProducer: [com.rnmapboxglexample/com.rnmapboxglexample.MainActivity](this:0x7f83bca800,id:1314,api:1,p:30674,c:265) queueBuffer: fps=0.53 dur=1901.81 max=1901.81 min=1901.81
09-29 00:02:00.370 ? D/SoLoader: libimagepipeline.so not found on /data/data/com.rnmapboxglexample/lib-main
09-29 00:02:00.370 ? D/SoLoader: libimagepipeline.so found on /data/app/com.rnmapboxglexample-2/lib/arm64
09-29 00:02:00.385 ? A/libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0x7fe570bc44 in tid 30674 (mapboxglexample)
```

**Comment by RobertSasak** [comment]
I solved the second issue. `<Image>` and `<ImageBackground>` cannot be used without source prop. This happen in _react-native-elements_ library in Header. This issue only occur on old Android phones. https://github.com/react-native-training/react-native-elements/issues/1938

**Comment by tsemerad** [comment]
Looks like adding `lintOptions { checkReleaseBuilds false }` didn't fix it. So I'm assuming the android build is building fine for you on your development machine? Maybe it has to do with the CI. I noticed that the Android build is being built on Linux, whereas I'm assuming you're on a Mac. @mfazekas , do you know why we're building it on Linux rather than Mac?

Also, I struggled with a PR earlier until I realized that the Travis CI was using yarn rather than npm to install things, so it's probably a good idea to install this project using yarn to avoid any discrepancies (although I doubt this is causing your issue).

**Comment by tsemerad** [comment]
Also, I noticed that the react-native package in the root `package.json` is still on 0.59, so that should be updated too.

**Comment by RobertSasak** [comment]
Yes, turning off linter did not help. As expected.

**Comment by tsemerad** [comment]
Oh! I didn't realize it was related to jetify. I think we just need to jetify the dep on postinstall (that's what I had to do on my personal project).

Just add the `jetifier` module to the example project's `package.json`, and add `&& npx jetifier` to its `postinstall` script.

**Comment by RobertSasak** [comment]
It was the jetifier. I find it strange that it not throw any error on my MacOS. 
Is there anything I can do before merge?

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
   "peerDependencies": {
     "prop-types": ">=15.5.8",
     "react": "^16.6.1",
-    "react-native": ">=0.57.6"
```
[the remark this answers is not in the document]
Comment (RobertSasak) [reply package.json]: My bad

File: android/rctmgl/build.gradle [hunk 2]
```diff
@@ -21,6 +21,9 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
+    lintOptions {
+        checkReleaseBuilds false
+    }
```
[the remark this answers is not in the document]
Comment (RobertSasak) [reply android/rctmgl/build.gradle]: Fixed

File: package.json [hunk 3]
```diff
@@ -76,7 +76,7 @@
     "prettier": "^1.11.1",
     "react": "16.8.6",
     "react-docgen": "^5.0.0-beta.1",
-    "react-native": "0.59.10",
+    "react-native": "0.60.0",
```
Comment (tsemerad) [inline package.json]: I was thinking this version should probably be v0.61.1 like in the example, but now that I look into it, I'm not sure what this react-native is being used for. It is a devDependency, but I'm not sure what for. I'd either set it to v0.61.1, or remove it if it really isn't needed. Sorry for not clarifying earlier.

Comment (RobertSasak) [reply package.json]: I am really happy that you helping me out. I am learning a lot. You are right, react-native is not a developer dependency. It is only needed when module is running and is already specified as peer dependency.

**Comment by tsemerad** [comment]
It looks like this react-native is needed for the unit tests. If you run `npm run unittest`, you'll see that it's failing. It also fails if you upgrade `react-native` to the latest version, so I'd just return it to v0.59.10 for now. The unit tests can be updated in a separate PR. Sorry for recommending upgrading that `react-native` version. Now we know what it is used for.

**Comment by RobertSasak** [comment]
I restored react-native in dev dependency. I guess test should be included in build.

**Comment by RobertSasak** [comment]
Is there anything else I can do?

**Comment by kristfal** [comment]
Hey RobertSasak! Thanks a lot for this. I'll have a look soon (and possibly push this all the way up to RN61.4).

**Comment by RichardLindhout** [comment]
ESLint couldn't find the config "@react-native-community" to extend from. Please check that the name of the config is correct.

**Comment by RobertSasak** [comment]
@RichardLindhout  I am not sure what exactly is the issue? Can you please explain it in more detail. I found this repo for the reference https://github.com/facebook/react-native/tree/master/packages/eslint-config-react-native-community

**Comment by RichardLindhout** [comment]
I don't know exactly either. Maybe missing in the package.json I have it working now, so it's fine.
https://github.com/react-native-mapbox-gl/maps/pull/546/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R52

**Pull request merged**